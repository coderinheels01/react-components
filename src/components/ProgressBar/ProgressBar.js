import React from "react";
import PropTypes from "prop-types";

/** Progress bar that display progress*/
const ProgressBar = ({ percent, width, height = 15 }) => (
  <div
    style={{
      border: "1px solid #d3d3d3",
      width: width,
      backgroundColor: "#898787"
    }}
  >
    <div
      style={{
        width: ProgressBar.calculateCompletionWidth(width, percent),
        height,
        backgroundColor: ProgressBar.calculateCompletionColor(percent)
      }}
    />
    <div />
  </div>
);

/** Calculate completed progress */
ProgressBar.calculateCompletionWidth = (width = 0, percent = 0) => {
  return parseInt(width * (percent / 100), 10);
};

/** Calculate progress bar color  */
ProgressBar.calculateCompletionColor = percent => {
  debugger;
  if (percent === 100) return "#36a01d";
  return percent > 50 ? "#72e356" : "#e02d14";
};

ProgressBar.propTypes = {
  /** Percentage of the progress completed*/
  percent: PropTypes.number.isRequired,

  /** Width of the progress bar*/
  width: PropTypes.number.isRequired,

  /** Height of the progress bar*/
  height: PropTypes.number
};

export default ProgressBar;
