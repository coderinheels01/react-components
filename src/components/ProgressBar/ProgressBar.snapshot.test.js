import React from "react";
import renderer from "react-test-renderer";
import ProgressBar from "./ProgressBar";

describe("ProgressBar", () => {
  let props;
  let progressBarTree;

  beforeEach(() => {
    props = {
      percent: 50,
      width: 100,
      height: 20
    };
    progressBarTree = renderer.create(<ProgressBar {...props} />);
  });

  it("render ProgressBar with 50 percent, 100 width and 20 height", () => {
    expect(progressBarTree).toMatchSnapshot();
  });
});
