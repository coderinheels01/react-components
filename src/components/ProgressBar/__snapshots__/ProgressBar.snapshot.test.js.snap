// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar render ProgressBar with 50 percent, 100 width and 20 height 1`] = `
<div
  className="sc-bdfBwQ ePcQmM"
  width={100}
>
  <div
    className="sc-gsTCUz gJDOts"
    height={20}
    width={50}
  />
</div>
`;
